File: remove-duplicates-from-sorted-array-ii/RemoveDuplicates.cs
public class Solution
{
    public int RemoveDuplicates(int[] nums)
    {
        var result = 0;
        var pointer = 0;
        var previous = int.MinValue;
        var count = 0;

        foreach (var num in nums)
        {
            if (num != previous || count < 2)
            {
                nums[pointer++] = num;
                result++;
                if (num != previous)
                {
                    previous = num;
                    count = 0;
                }
            }
            count++;
        }

        return result;
    }
}

File: remove-duplicates-from-sorted-array-ii/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let result = 0,
    pointer = 0,
    count = 0,
    previous;
  nums.forEach((num) => {
    if (num != previous || count < 2) {
      nums[pointer++] = num;
      if (num != previous) {
        previous = num;
        count = 0;
      }
      result++;
    }
    count++;
  });
  return result;
};
